wolfssl/wolfcrypt/ext_lms.h

```diff
@@ -0,0 +1,63 @@
+/* ext_lms.h
+ *
+ * Copyright (C) 2006-2023 wolfSSL Inc.
+ *
+ * This file is part of wolfSSL.
+ *
+ * wolfSSL is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfSSL is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+
+#ifndef EXT_LMS_H
+#define EXT_LMS_H
+
+#ifdef WOLFSSL_HAVE_LMS
+#include <wolfssl/wolfcrypt/lms.h>
+
+#if !defined(HAVE_LIBLMS)
+#error "This code requires liblms"
+#endif
+
+/* hash-sigs LMS HSS includes */
+#include <hss.h>
+
+#if defined(WOLFSSL_WC_LMS)
+#error "This code is incompatible with wolfCrypt's implementation of LMS."
```

Comment:
Trying the configure `./configure --enable-lms --with-liblms=/home/jak/Documents/hash_sigs/src/` looks to run successfully. I think configure should error out if having both enabled at the same time is invalid.

Though it looks to have silently removed the --enable-lms:

```
   * LMS:                        yes
   * LMS wolfSSL impl:           no
```

Reply:
> Trying the configure `./configure --enable-lms --with-liblms=/home/jak/Documents/hash_sigs/src/` looks to run successfully. I think configure should error out if having both enabled at the same time is invalid.
> 
> Though it looks to have silently removed the --enable-lms:
> 
> ```
>    * LMS:                        yes
>    * LMS wolfSSL impl:           no
> ```

That's actually ok. The conflict would be
```
./configure --enable-lms=wolfssl --with-liblms=/home/jordan/hash_sigs/src
```

This would enable the wolfcrypt version of LMS while trying to enable our use of an external lms with `with-liblms`.

Reply:
Great. User error on my end, it does catch the case of both defined at build time ok:

```
make[1]: Entering directory '/home/jak/Documents/wolfssl'
make[2]: Entering directory '/home/jak/Documents/wolfssl'
make[2]: warning: -j5 forced in submake: resetting jobserver mode.
  CC       wolfcrypt/benchmark/benchmark.o
  CC       wolfcrypt/src/src_libwolfssl_la-hmac.lo
  CC       wolfcrypt/src/src_libwolfssl_la-hash.lo
  CC       wolfcrypt/src/src_libwolfssl_la-cpuid.lo
  CC       wolfcrypt/src/src_libwolfssl_la-kdf.lo
In file included from wolfcrypt/benchmark/benchmark.c:160:
./wolfssl/wolfcrypt/ext_lms.h:36:2: error: #error "This code is incompatible with wolfCrypt's implementation of LMS."
   36 | #error "This code is incompatible with wolfCrypt's implementation of LMS."
      |  ^~~~~
make[2]: *** [Makefile:6073: wolfcrypt/benchmark/benchmark.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: Leaving directory '/home/jak/Documents/wolfssl'
make[1]: *** [Makefile:7672: all-recursive] Error 1
make[1]: Leaving directory '/home/jak/Documents/wolfssl'
make: *** [Makefile:4569: all] Error 2
```


```
   * LMS:                        wolfssl
   * LMS wolfSSL impl:           yes
```